### PR TITLE
Refactor:Remove GlobalV in gint

### PR DESCRIPTION
**Pull request by A-006**
1. Remove global variables in gint and replace them with parameter passing.
2. Some header file dependencies have been removed, so remove them.
3.The setting of omp parallel in gint is not reasonable; it should be modified to an appropriate location. 
4. Replace the use of new and delete in vindex with vector, ensuring no memory leaks while maintaining performance.

### CODE REVIEW

File: source/module_lr/esolver_lrtd_lcao.cpp [hunk 1]
```diff
@@ -326,13 +328,17 @@ LR::ESolver_LR<T, TR>::ESolver_LR(const Input_para& inp, UnitCell& ucell) : inpu
         psi_u,
         dpsi_u,
         d2psi_u,
-        GlobalV::NUM_STREAM);
-    psi_u.clear();
```
Comment (mohanchen) [inline source/module_lr/esolver_lrtd_lcao.cpp:330]: why did you delete psi_u.clear() and the following lines?

File: source/module_esolver/esolver_ks_lcao_elec.cpp [hunk 2]
```diff
@@ -90,13 +90,17 @@ void ESolver_KS_LCAO<TK, TR>::set_matrix_grid(Record_adj& ra)
                              psi_u,
                              dpsi_u,
                              d2psi_u,
-                             GlobalV::NUM_STREAM);
-    psi_u.clear();
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao_elec.cpp:94]: why did you delete lines realted to psi_u, etc.?

File: source/module_lr/esolver_lrtd_lcao.cpp (hunk 1 above)
Comment (A-006) [reply source/module_lr/esolver_lrtd_lcao.cpp:330]: `psi_u` is created within this function. Since it is of vector type, it is automatically released when the function ends, so there is no need for manual memory management.

File: source/module_esolver/esolver_ks_lcao_elec.cpp (hunk 2 above)
Comment (A-006) [reply source/module_esolver/esolver_ks_lcao_elec.cpp:94]: `psi_u` is created within this function. Since it is of vector type, it is automatically released when the function ends, so there is no need for manual memory management.

Comment (mohanchen) [reply source/module_esolver/esolver_ks_lcao_elec.cpp:94]: When you look into the code, it's very strange. Talk to you later.

File: source/module_esolver/esolver_ks_lcao_elec.cpp [hunk 3]
```diff
@@ -90,13 +90,17 @@ void ESolver_KS_LCAO<TK, TR>::set_matrix_grid(Record_adj& ra)
...
-                             PARAM.inp.nstream);
-    psi_u.clear();
-    psi_u.shrink_to_fit();
-    dpsi_u.clear();
-    dpsi_u.shrink_to_fit();
-    d2psi_u.clear();
-    d2psi_u.shrink_to_fit();
+                             PARAM.inp.nstream,
+                             GlobalV::GAMMA_ONLY_LOCAL,
+                             GlobalV::NSPIN,
```
Comment (mohanchen) [inline source/module_esolver/esolver_ks_lcao_elec.cpp]: use PARAM.inp.xxx instead of GlobalV

File: source/module_hamilt_lcao/module_gint/grid_technique.cpp [hunk 4]
```diff
@@ -77,7 +68,24 @@ void Grid_Technique::set_pbc_grid(
...
     this->dpsi_u = dpsi_u;
     this->d2psi_u = d2psi_u;
-
+    this->gamma_only_local=gamma_only_local;
+    this->nspin = nspin;
+    this->domag = domag;
+    this->npol = ucell.get_npol();
+    this->nlocal = nlocal;
+
+    int nproc;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/grid_technique.cpp]: initialize the variable

File: source/module_esolver/esolver_fp.cpp [hunk 5]
```diff
@@ -305,4 +306,51 @@ void ESolver_FP::print_rhofft(const Input_para& inp, std::ofstream& ofs)
     }
 }
 
+#ifdef __LCAO
+void ESolver_FP::init_orb(double& dr_uniform,
```
Comment (mohanchen) [inline source/module_esolver/esolver_fp.cpp]: Please don't add new functions in ESolver_FP

File: source/module_esolver/esolver_fp.h [hunk 6]
```diff
@@ -59,6 +59,15 @@ namespace ModuleESolver
         //! K points in Brillouin zone
         K_Vectors kv;
 
+#ifdef __LCAO
```
Comment (mohanchen) [inline source/module_esolver/esolver_fp.h]: Please don't add new functions in ESolver_FP

Comment (A-006) [reply source/module_esolver/esolver_fp.h]: init_orb has been removed from ESolver_FP and placed in esolver/init_orb.hpp.

File: source/module_esolver/esolver_fp.cpp (hunk 5 above)
Comment (A-006) [reply source/module_esolver/esolver_fp.cpp]: init_orb has been removed from ESolver_FP and placed in esolver/init_orb.hpp.

**Comment by mohanchen** [comment]
You seem don't understand LCAO, I will close this PR. You can resubmit it later.
